src/pyodide/internal/snapshot.ts

```diff
@@ -264,8 +264,13 @@ function memorySnapshotDoImports(Module: Module): string[] {
         MetadataReader.getWorkerFiles('py')
       )
     );
-
-  const deduplicatedModules = [...new Set(importedModules)];
+  const moduleSet = new Set(importedModules);
+  // Importing pyodide before `finalizeBootstrap()` is called causes mayhem
+  // Make sure we don't do that!
+  // TODO: in Pyodide 0.27 we can call finalizeBootstrap() before taking the snapshot. Doing so will
```

Comment:
Perhaps we need a way to check pyodide version when generating the bundles so we could make things like this conditional

Reply:
Indeed we do.